jmdaemon/jmdaemon/onionmc.py

```diff
@@ -841,8 +843,8 @@ def forward_privmsg_to_peer(self, nick: str, message: str,
             return
         # The `message` passed in has format COMMAND_PREFIX||command||" "||msg
         # we need to parse out cmd, message for sending.
-        _, cmdmsg = message.split(COMMAND_PREFIX)
-        cmdmsglist = cmdmsg.split(" ")
+        cmdsmsgs = COMMAND_PREFIX.join(message.split(COMMAND_PREFIX)[1:])
+        cmdmsglist = cmdsmsgs.split(" ")

```

Comment:
> Have we split and joined only to remove the COMMAND_PREFIX at the beginning?

Yes but that was intended. Specifically, what was intended was that this:

`!sw0reloffer a b c d !sw0reloffer a b c d ...`

becomes:

`sw0reloffer a b c d !sw0reloffer a b c d ...`

i.e. it does *not* split further on the next `!`.

This means that `cmd` becomes specifically `sw0reloffer`.
And `msg` becomes specifically `a b c d !sw0reloffer a b c d ...`

which is what we need it to be to pass into `get_privmsg`.

I don't think anything is wrong here (indeed I'm a bit confused because I was quite sure that that's what you, correctly, observed as the mistake in the first version).

Reply:
Ohh, sorry 1. is a brain fart.
re: 2. I'm confused why you choose this way to do it if we just want to remove the prefix from the beginning. I.e., Can't we do `cmdsmsgs = message[len(COMMAND_PREFIX):]` or something faster and simpler along those lines?

Reply:
Probably yeah. I don't imagine there's a scenario where just lopping off the front is wrong. But it's OK like this imo.

Reply:
IMHO if we want to keep split, I think then we should do like `cmdsmsgs = message.split(COMMAND_PREFIX, 1)[1]`. The second argument is the maximum number of splits to do (or 1 in this case, which is exactly what we want), this is faster then current code and it seems much cleaner to me while maintaining the same behavior.

Reply:
OK thanks done in https://github.com/JoinMarket-Org/joinmarket-clientserver/pull/1182/commits/2966985b9104905b9fad263342e21a608a4fdf5e